app/models/ems_event.rb

```diff
@@ -1,5 +1,6 @@
 class EmsEvent < EventStream
   include_concern 'Automate'
+  supports :timeline
```

Comment:
I didn't know what to call this feature so I went with timeline.  Basically, we need to ensure the other EventStream subclasses, don't show up in the payload until we support them for timelines.

Reply:
timeline seems fine - @agrare opinions?

It's a little weird to be using supports for this, because it's not really a provider feature, but I think I'm ok with that.

Reply:
I'm not against using supports for non provider features.  If this is just a single subclass check though could you just use `EmsEvent` for the query?

Reply:
yeah, supports is really about the _outside callers_ (like UI and API) having an interface for asking if a feature is supported and being able to filter on it, in addition to _why_ it's not supported for presentation putposes.  In this case, that's not what it's being used for, as we aren't going to be presenting anything in the UI for the non-supported classes.

I'm thinking this is much cleaner with a simpler `subclasses.select { respond_to? }` construct and only grab the subclasses that present a `group_levels_and_details` function.

Reply:
See my comment here too: https://github.com/ManageIQ/manageiq/pull/21552/files#r747633985

Reply:
Note, the motivation was something different than what the UI was doing, namely: 

```ruby
record.has_events? || record.has_events?(:policy_events) # EmsEvent || MiqEvent
```

```
joerafaniello@Joes-MBP-2 manageiq-ui-classic % git grep has_events
app/helpers/application_helper/button/availability_zone_timeline.rb:    @error_message = _("No Timeline data has been collected for this Availability Zone") unless @record.has_events?
app/helpers/application_helper/button/container_timeline.rb:    @record.has_events? || @record.has_events?(:policy_events)
app/helpers/application_helper/button/ems_cluster_timeline.rb:    @error_message = _("No Timeline data has been collected for this Cluster") unless @record.has_events? || @record.has_events?(:policy_events)
app/helpers/application_helper/button/ems_storage_timeline.rb:    unless @record.has_events? || @record.has_events?(:policy_events)
app/helpers/application_helper/button/ems_timeline.rb:    unless @record.has_events? || @record.has_events?(:policy_events)
app/helpers/application_helper/button/host_timeline.rb:    @error_message = _("No Timeline data has been collected for this Host") unless @record.has_events? || @record.has_events?(:policy_events)
app/helpers/application_helper/button/miq_template_timeline.rb:    unless @record.has_events? || @record.has_events?(:policy_events)
app/helpers/application_helper/button/physical_server_timeline.rb:    unless @record.has_events?
app/helpers/application_helper/button/vm_timeline.rb:    @record.has_events? || @record.has_events?(:policy_events)
```

Reply:
Granted, the UI is just checking for events for display purposes but obviously, if they wanted to display CustomEvent, CustomButtonEvent, or RequestEvent, these conditionals would grow gnarly.

Reply:
For the backend code, we don't care about having events for display purposes.  We only want to provide the details about options for classes that support timelines.  🤷

Reply:
> I'm thinking this is much cleaner with a simpler `subclasses.select { |e| e.respond_to?(:group_levels_and_details) }` construct.

I guess I can try doing this in `EventStream`

```ruby
def self.timeline_classes
  subclasses.select { |e| e.respond_to?(:group_levels_and_details) }
end
```

I can worry about what the UI does with checking if there are events for record later.

Reply:
This should be done, I dropped the private method and simplified things.